torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -256,25 +257,43 @@ bool IsUninitializedNode(Node* n) {
 
 // Infer shape and type of the uninitialized_output from the corresponding
 // output of the other subblock. prim::Uninitialized node is proven to be
-// unused. So replace this node with a constant of the inferred shape and type.
+// unused. So replace this node with a Constant (TensorType) or
+// Sequence (ListType) of the inferred shape and type.
 void InferShapeTypeForUninitializedOutput(
     Graph* graph,
     Block* block,
     Value* uninitialized_output,
     Value* other_output) {
-  auto output_type = other_output->type()->expect<TensorType>();
-  auto elem_type = at::initialTensorOptions().dtype(output_type->scalarType());
-  Node* const_node = graph->create(::c10::onnx::Constant, 1);
-
-  if (output_type->sizes().concrete_sizes().has_value()) {
-    auto size = output_type->sizes().concrete_sizes().value();
-    const_node->t_(attr::value, at::zeros(size, elem_type));
-    const_node->output()->setType(other_output->type());
-    const_node->output()->copyMetadata(other_output);
-  } else {
-    const_node->t_(attr::value, at::zeros({}, elem_type));
-    const_node->output()->setType(
-        TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+  Node* const_node = nullptr;
+  if (auto output_type = other_output->type()->cast<TensorType>()) {
+    auto elem_type =
+        at::initialTensorOptions().dtype(output_type->scalarType());
+    const_node = graph->create(::c10::onnx::Constant, 1);
+
+    if (output_type->sizes().concrete_sizes().has_value()) {
+      auto size = output_type->sizes().concrete_sizes().value();
+      const_node->t_(attr::value, at::zeros(size, elem_type));
+      const_node->output()->setType(other_output->type());
+    } else {
+      const_node->t_(attr::value, at::zeros({}, elem_type));
+      const_node->output()->setType(
+          TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+    }
+  } else if (auto output_type = other_output->type()->cast<ListType>()) {
+    TypePtr elem = output_type->getElementType();
+    const_node = graph->create(::c10::onnx::SequenceEmpty, 1);
+    if (elem->cast<TensorType>() &&
+        elem->cast<TensorType>()->scalarType().has_value()) {
+      auto scalar_type = elem->cast<TensorType>()->scalarType().value();
+      auto onnx_type = ATenTypeToOnnxType(scalar_type);
+      const_node->i_(attr::dtype, onnx_type);
+      const_node->output()->setType(other_output->type());
+    } else {
+      std::cerr
+          << "Warning: UninitializedOutput - Invalid elem Type of ListTensor found."
+          << std::endl;
+      const_node->output()->setType(other_output->type());
+    }
   }
   const_node->insertBefore(block->return_node());
```

Comment:
Probably we need add shape inference for this const node, like
```cpp
const ParamMap empty_params_dict = {};
auto opset_version =
        py::cast<int>(onnx_symbolic.attr("_export_onnx_opset_version"));
ONNXShapeTypeInference(const_node, empty_params_dict, opset_version);
```
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/onnx.cpp#L286

Reply:
> Probably we need add shape inference for this const node, like
> 
> ```c++
> const ParamMap empty_params_dict = {};
> auto opset_version =
>         py::cast<int>(onnx_symbolic.attr("_export_onnx_opset_version"));
> ONNXShapeTypeInference(const_node, empty_params_dict, opset_version);
> ```
> 
> https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/onnx.cpp#L286

Done, shape inference needed after `constant` node.

Reply:
nit: It is better to move `const_node->insertBefore(block->return_node());` before shape inference logic. (although for const node it may be fine)